playbooks/ngc-ready.yml

```diff
@@ -0,0 +1,137 @@
+---
+
+- include: bootstrap-python.yml
+- include: bootstrap-ssh.yml
+- include: bootstrap-sudo.yml
+
+- include: docker.yml
+- include: nvidia-driver.yml
+- include: nvidia-docker.yml
+
+- hosts: all
+  become: true
+  vars:
+    ngc_container_tag: 18.10-py3
+  tasks:
+    - name: install docker python module
+      apt:
+        name: python-docker
+        state: present
+        update_cache: yes
+      when: ansible_distribution == "Ubuntu"
+    - name: Enable EPEL repo
+      package:
+        name:
+          - epel-release
+        state: present
+      when: ansible_os_family == 'RedHat'
+    - name: Install package dependencies
+      package:
+        name:
+          - python-pip
+        state: present
+      when: ansible_os_family == "RedHat"
+    - name: Install python dependencies
+      pip:
+        name:
+          - docker
+        state: present
+      when: ansible_os_family == "RedHat"
+    - name: pull CUDA container
+      docker_container:
+        name: gpu-test-pull
+        image: nvcr.io/nvidia/cuda
+        auto_remove: yes
+        pull: yes
+      tags:
+        - test
+        - never
+    - name: test CUDA container
+      docker_container:
+        name: gpu-test
+        image: nvcr.io/nvidia/cuda
+        detach: no
+        command: nvidia-smi -L
+      register: cuda
+      tags:
+        - test
+        - never
+    - name: functional test output
+      debug:
+        var: cuda.ansible_facts.docker_container.Output
+      tags:
+        - test
+        - never
+    - name: remove CUDA container
+      docker_container:
+        name: gpu-test
+        state: absent
+      tags:
+        - test
+        - never
+    - name: pull pytorch container
+      docker_container:
+        name: pytorch-pull
+        image: "nvcr.io/nvidia/pytorch:{{ ngc_container_tag }}"
+        auto_remove: yes
+        pull: yes
+      tags:
+        - test
+        - never
+    - name: test pytorch
+      docker_container:
+        name: pytorch
+        image: "nvcr.io/nvidia/pytorch:{{ ngc_container_tag }}"
+        working_dir: /opt/pytorch/examples/upstream/mnist
+        detach: no
```

Comment:
Would it hurt to add the following? Otherwise it won't work on virtual clusters, for example.

```
network_mode: host
```

Reply:
The tests don't require any networking so it's probably not necessary

Reply:
They do for some - for example the pytorch one fails if you can't download the MNIST dataset.

Reply:
ok, done